### PR TITLE
Feat: Jump To Screen Functionality

### PR DESCRIPTION
**Pull request by jeremiahseun**
This PR includes the functionality for Jump To Screen view in which the user searches for recent users and contacted users.
This is the introductory aspect of this PR. As the endpoint grows, the implementation will be updated.

Other minor corrections are made.

### CODE REVIEW

File: lib/package/base/jump_to_request/jump_to_api.dart [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+      '/api/v1/6145ad38285e4a18402073ad/channels/users/$currentUserId/';
+
+  Map<String, dynamic> headers = {
+    'Authorization':
+        'Bearer ${storageService.getString(StorageKeys.currentSessionToken)}'
+  };
+
+  /// Fetches a list of  all channels in that organization
+  Future<List<ChannelsSearch>> allChannelsList() async {
+    print(storageService.getString(StorageKeys.currentSessionToken));
```
Comment (isaacadariku) [inline lib/package/base/jump_to_request/jump_to_api.dart:25]: Remove the print statement and make use of stacked_logger

`log.i()`

File: lib/package/base/jump_to_request/jump_to_api.dart [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    final response = await _channelsApi.get(allChannelsPath, headers: headers);
+    return (response!.data?['data'] as List)
+        .map((e) => ChannelsSearch.fromJson(e))
+        .toList();
+  }
+
+  /// Fetches a list of channels that a user is, in that organization
+  Future<List<ChannelsSearch>> joinedChannelsList() async {
+    print(storageService.getString(StorageKeys.currentSessionToken));
```
Comment (isaacadariku) [inline lib/package/base/jump_to_request/jump_to_api.dart:34]: Here too

File: lib/package/base/jump_to_request/jump_to_api.dart [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  }
+
+  /// Fetches a list of members in that organization
+  Future<List<UserSearch>> fetchListOfMembers() async {
+    final response = await _dmApi
+        .get('organizations/6145ad38285e4a18402073ad/members/', headers: {
+      'Authorization':
+          'Bearer ${storageService.getString(StorageKeys.currentSessionToken)}'
+    });
+    print("Org members length - ${response?.data?['data'].length}");
```
Comment (isaacadariku) [inline lib/package/base/jump_to_request/jump_to_api.dart:49]: Here too

Comment (isaacadariku) [inline lib/package/base/jump_to_request/jump_to_api.dart:49]: Here too

File: lib/ui/view/dm_chat_view/dm_jump_to_viewmodel.dart [hunk 4]
```diff
@@ -1,7 +1,40 @@
...
+        AppToast.instance.message(null, 'Check your internet connection');
+        return;
+      }
+      setBusy(true);
+      userSearch = await api.fetchListOfMembers();
+      joinedChannelsSearch = await api.joinedChannelsList();
+      allChannelsSearch = await api.allChannelsList();
+      setBusy(false);
+    } catch (e) {
+      print(e.toString());
```
Comment (isaacadariku) [inline lib/ui/view/dm_chat_view/dm_jump_to_viewmodel.dart:34]: Use stacked_logger

File: lib/ui/view/dm_chat_view/dm_jump_to_viewmodel.dart [hunk 5]
```diff
@@ -1,7 +1,40 @@
...
+      AppToast.instance.error(null, 'Error Occured');
+    }
+  }
+
+  final String currentOrgId = '6145ad38285e4a18402073ad';
```
Comment (isaacadariku) [inline lib/ui/view/dm_chat_view/dm_jump_to_viewmodel.dart:39]: Why is this a static value?

**Review by isaacadariku — CHANGES_REQUESTED** [review]
Make the required changes

File: lib/package/base/jump_to_request/jump_to_api.dart (hunk 1 above)
Comment (jeremiahseun) [reply lib/package/base/jump_to_request/jump_to_api.dart]: Changes made

File: lib/package/base/jump_to_request/jump_to_api.dart (hunk 2 above)
Comment (jeremiahseun) [reply lib/package/base/jump_to_request/jump_to_api.dart]: Changes made

File: lib/package/base/jump_to_request/jump_to_api.dart (hunk 3 above)
Comment (jeremiahseun) [reply lib/package/base/jump_to_request/jump_to_api.dart]: Changes made

Comment (jeremiahseun) [reply lib/package/base/jump_to_request/jump_to_api.dart]: Changes made

File: lib/ui/view/dm_chat_view/dm_jump_to_viewmodel.dart (hunk 4 above)
Comment (jeremiahseun) [reply lib/ui/view/dm_chat_view/dm_jump_to_viewmodel.dart]: Changes made

File: lib/ui/view/dm_chat_view/dm_jump_to_viewmodel.dart [hunk 6]
```diff
@@ -31,7 +33,7 @@ class DmJumpToViewModel extends BaseViewModel {
       allChannelsSearch = await api.allChannelsList();
       setBusy(false);
     } catch (e) {
-      print(e.toString());
+      log.i(e.toString());
```
Comment (isaacadariku) [inline lib/ui/view/dm_chat_view/dm_jump_to_viewmodel.dart:36]: Use `log.e()` for errors

File: lib/ui/view/plugins/add_plugin_view.dart [hunk 7]
```diff
@@ -55,7 +57,7 @@ class AddPluginView extends StatelessWidget {
                         value: model.checked,
                         onChanged: (bool? value) {
                           model.setChecked(value!);
-                          print(value);
+                          log.i(value);
```
Comment (isaacadariku) [inline lib/ui/view/plugins/add_plugin_view.dart:60]: no need for this log, you can remove it. No logic in the UI

File: lib/ui/view/preference/preference_viewmodel.dart [hunk 8]
```diff
@@ -27,7 +29,7 @@ class PreferenceViewModel extends BaseViewModel {
     );
 
     if (dialogResult != null && dialogResult.confirmed == true) {
-      print(dialogResult.data);
+      log.i(dialogResult.data);
```
Comment (isaacadariku) [inline lib/ui/view/preference/preference_viewmodel.dart:32]: This is not needed too

File: lib/ui/view/profile_page/profile_page_viewmodel.dart [hunk 9]
```diff
@@ -25,8 +27,8 @@ class ProfilePageViewModel extends BaseViewModel {
         // the username should be replaced with the username from the rest api
         name: 'User Name',
       );
-      print(result);
-      print(response.statusCode);
+      log.i(result);
+      log.i(response.statusCode);
```
Comment (isaacadariku) [inline lib/ui/view/profile_page/profile_page_viewmodel.dart:31]: I think dio have a log already base on the response, no need to have a log in the viewmodel

File: lib/ui/view/sign_up/sign_up_view.dart [hunk 10]
```diff
@@ -178,7 +180,7 @@ class SignUpView extends StatelessWidget with $SignUpView {
                           ),
                           child: InkWell(
                             onTap: () {
-                              print(' chiboy clicked');
+                              log.i(' chiboy clicked');
```
Comment (isaacadariku) [inline lib/ui/view/sign_up/sign_up_view.dart:183]: This too should be remove

File: lib/ui/view/workspace/email_confirmation/email_confirmation_view.dart [hunk 11]
```diff
@@ -47,7 +49,7 @@ To confirm your email address, tap the button in the email we sent to lukeskylwa
                         onSurface: HexColor('#00B87C'),
                       ),
                       onPressed: () {
-                        print('Pressed');
+                        log.i('Pressed');
```
Comment (isaacadariku) [inline lib/ui/view/workspace/email_confirmation/email_confirmation_view.dart:52]: here too

File: lib/ui/view/workspace/email_confirmation/email_confirmation_viewmodel.dart [hunk 12]
```diff
@@ -3,28 +3,30 @@ import 'dart:io';
...
+        log.i('App Email launched!');
       }).catchError((err) {
         // ignore: deprecated_member_use
         snackbar.showCustomSnackBar(
           duration: const Duration(seconds: 3),
           variant: SnackbarType.failure,
           message: 'App Email not found!',
         );
-        print(err);
+        log.i(err);
```
Comment (isaacadariku) [inline lib/ui/view/workspace/email_confirmation/email_confirmation_viewmodel.dart:29]: log.e() should be use

File: lib/ui/view/workspace/workspace_view/workspace_viewmodel.dart [hunk 13]
```diff
@@ -57,7 +58,7 @@ class WorkspaceViewModel extends BaseViewModel {
 
       setBusy(false);
     } catch (e) {
-      print(e.toString());
+      log.i(e.toString());
```
Comment (isaacadariku) [inline lib/ui/view/workspace/workspace_view/workspace_viewmodel.dart:61]: Here too

File: lib/ui/view/workspace/workspace_view/workspace_viewmodel.dart [hunk 14]
```diff
@@ -100,7 +101,7 @@ class WorkspaceViewModel extends BaseViewModel {
       navigation.replaceWith(Routes.navBarView);
       // navigation.popRepeated(1);
     } catch (e) {
-      print(e.toString());
+      log.i(e.toString());
```
Comment (isaacadariku) [inline lib/ui/view/workspace/workspace_view/workspace_viewmodel.dart:104]: Here too

**Review by isaacadariku — CHANGES_REQUESTED** [review]
Make the new changes and we are good to go

File: lib/ui/view/dm_chat_view/dm_jump_to_viewmodel.dart (hunk 6 above)
Comment (jeremiahseun) [reply lib/ui/view/dm_chat_view/dm_jump_to_viewmodel.dart:37]: Alright

File: lib/ui/view/plugins/add_plugin_view.dart (hunk 7 above)
Comment (jeremiahseun) [reply lib/ui/view/plugins/add_plugin_view.dart:60]: Alright

File: lib/ui/view/profile_page/profile_page_viewmodel.dart (hunk 9 above)
Comment (jeremiahseun) [reply lib/ui/view/profile_page/profile_page_viewmodel.dart:31]: Alright. I will check it out

File: lib/ui/view/sign_up/sign_up_view.dart (hunk 10 above)
Comment (jeremiahseun) [reply lib/ui/view/sign_up/sign_up_view.dart:183]: Alright

File: lib/ui/view/workspace/email_confirmation/email_confirmation_viewmodel.dart (hunk 12 above)
Comment (jeremiahseun) [reply lib/ui/view/workspace/email_confirmation/email_confirmation_viewmodel.dart:29]: Alright

File: lib/ui/view/workspace/workspace_view/workspace_viewmodel.dart (hunk 13 above)
Comment (jeremiahseun) [reply lib/ui/view/workspace/workspace_view/workspace_viewmodel.dart:61]: Alright

**Review by isaacadariku — APPROVED** [review]
LGTM! Weldone

**Pull request merged**